lib/membrane_mp4/container/parse_helper.ex

```diff
@@ -38,19 +39,33 @@ defmodule Membrane.MP4.Container.ParseHelper do
         box = %{content: content, size: content_size, header_size: header_size}
         parse_boxes(data, schema, context, [{name, box} | acc])
 
+      leftover: leftover ->
+        {:error, [box: name, reason: {:non_empty_leftover, leftover}]}
+
       try: {:error, context} ->
         {:error, [box: name] ++ context}
     end
   end
 
-  defp parse_fields(data, [], context) do
-    {:ok, {%{}, data}, context}
+  defp parse_fields(data, fields, context) do
```

Comment:
Couldn't we just do:
```suggestion
  defp parse_fields(data, fields, context, acc \\ %{}) do
```
put the `do_parse_fields` logic here and get rid of `do_parse_fields` seaparate function?
There is quite a lot of these `*_parse_field(s)?" functions :D

Reply:
I don't like this approach, as you can easily forget passing `acc`, which can lead to weird bugs